fe/parameterize_batch.py

```diff
@@ -0,0 +1,387 @@
+    fwd_sched = schedule[:T//2]
+    bkwd = all_du_dls[:, T//2:]
+    bkwd_sched = schedule[T//2:]
+    dG_fwd = math_utils.trapz(fwd, fwd_sched) # integral from inf to 0
+    dG_bkwd = math_utils.trapz(bkwd, bkwd_sched) # integral from 0 to inf
+    # dG_fwd and dG_bkwd have the same sign, so we need to flip dG_bkwd so the
+    # direction of integral is the same (requirement for pymbar.BAR)
+    dG_bkwd = -dG_bkwd # this is needed for BAR to be correct
+
+    # this is in kJ/mol, inputs to BAR needs to be in 1/kT.
+    kT = 2.479
+    dG_fwd /= kT
+    dG_bkwd /= kT
+    pred_dG = loss.mybar(jnp.stack([dG_fwd, dG_bkwd]))
+    pred_dG *= kT
+
+    return jnp.abs(pred_dG - true_dG)
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description='Quick Test')
+    parser.add_argument('--frames_dir', type=str, required=True)
+    parser.add_argument('--precision', type=str, required=True)    
+    parser.add_argument('--complex_pdb', type=str, required=True)
+    parser.add_argument('--ligand_sdf_file', type=str, nargs='*')
+    parser.add_argument('--num_gpus', type=int, required=True)
+    parser.add_argument('--jobs_per_gpu', type=int, required=True)
+    parser.add_argument('--num_conformers', type=int, required=True)
+    parser.add_argument('--true_dG_file', type=str, nargs='*')
+
+    args = parser.parse_args()
+
+    assert os.path.isdir(args.frames_dir)
+
+    true_dG_list = []
+    for true_dG_file in args.true_dG_file:
+        f = open(true_dG_file, 'r').readlines()
+        for line in f:
+            true_dG = float(line.split()[0])
+            true_dG_list.append(true_dG)
+
+    if args.precision == 'single':
+        precision = np.float32
+    elif args.precision == 'double':
+        precision = np.float64
+    else:
+        raise Exception("precision must be either single or double")
+
+    num_gpus = args.num_gpus
+    num_workers = args.num_gpus*args.jobs_per_gpu
+
+    print('Creating multiprocessing pool with',args.num_gpus, 'gpus and', args.jobs_per_gpu, 'jobs per gpu')
+    pool = multiprocessing.Pool(num_workers)
+
+    start = 1e3
+    end = 1.0
+    NT = 500
+    base = np.exp(np.log(end/start)/NT)
+    exps = np.arange(NT)
+    part_one = np.power(base, exps)*start
+    part_two = np.linspace(1.0, 0.3, 1000)
+    part_three = np.linspace(0.3, 0.0, 4000)
+
+    forward_schedule = np.concatenate([part_one, part_two, part_three])
+    backward_schedule = forward_schedule[::-1]
+    lambda_schedule = np.concatenate([forward_schedule, backward_schedule])
+
+    T = lambda_schedule.shape[0]
+    assert T % 2 == 0
+    dt = 0.001
+    step_sizes = np.ones(T)*dt
+    #T_min = 500
+    assert T % 2 == 0
+    cas = np.ones(T)*0.93
+    epoch = 0
+    lr = 5e-4
+
+    ligand_list = []
+    for ligand_sdf_file in args.ligand_sdf_file:
+        f = open(ligand_sdf_file, 'r').readlines()
+        for file_name in f:
+            ligand_list.append(Chem.SDMolSupplier(file_name.rstrip('\n'), removeHs=False))
```

Comment:
I would make a single ligand sdf file with all the molecules, and attach the true_dG as a property. Then feed in a tuple of (mol, true_dG) as elements of data.